resources/datalake/resource_aws_datalake.go

```diff
@@ -404,9 +406,11 @@ func (r *awsDatalakeResource) Delete(ctx context.Context, req resource.DeleteReq
 		return
 	}
 
-	if err := waitForDatalakeToBeDeleted(ctx, state.DatalakeName.ValueString(), time.Hour, r.client.Datalake, state.PollingOptions); err != nil {
```

Comment:
shouldn't we delete the `waitForDatalakeToBeDeleted` method (and the rest of `waitFor*` ones to reduce the leftover codes?

Reply:
We are still using them if the async attribute is false